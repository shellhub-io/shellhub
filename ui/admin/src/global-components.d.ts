// src/global-components.d.ts
import SnackbarComponent from "./components/Snackbar/Snackbar.vue";

declare module "@vue/runtime-core" {
  export interface GlobalComponents {
    SnackbarComponent: typeof SnackbarComponent;
    RouterLink: typeof import("vue-router")["RouterLink"];
    RouterView: typeof import("vue-router")["RouterView"];
  }
}
