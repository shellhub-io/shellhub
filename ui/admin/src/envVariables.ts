/* eslint-disable import/prefer-default-export */
export const envVariables = {
  announcementsEnable: process.env.SHELLHUB_ANNOUNCEMENTS === "true",
  tinyMceKey: process.env.TINYMCE_API_KEY,
};
