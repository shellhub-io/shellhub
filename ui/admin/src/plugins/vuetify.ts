// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";
import { VNumberInput } from "vuetify/labs/VNumberInput";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
  components: {
    VNumberInput,
  },
});
