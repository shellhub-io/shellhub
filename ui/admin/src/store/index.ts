import { InjectionKey } from "vue";
import { createStore, Store, useStore as vuexUseStore } from "vuex";
import { stats, StatsState } from "./modules/stats";

import { UsersState, users } from "./modules/users";
import { AuthState, auth } from "./modules/auth";
import { DevicesState, devices } from "./modules/devices";
import { firewallRules, FirewallRulesState } from "./modules/firewall_rules";
import { namespaces, NamespacesState } from "./modules/namespaces";
import { layout, LayoutState } from "./modules/layout";
import { license, LicenseState } from "./modules/license";
import { publicKeys, PublicKeysState } from "./modules/public_keys";
import { sessions, SessionsState } from "./modules/sessions";
import { snackbar, SnackbarState } from "./modules/snackbar";
import { spinner, SpinnerState } from "./modules/spinner";
import { announcement, AnnouncementState } from "./modules/announcement";
import { instance, InstanceState } from "./modules/instance";

export interface State {
  user: UsersState;
  auth: AuthState;
  devices: DevicesState;
  firewallRules: FirewallRulesState;
  layout: LayoutState;
  license: LicenseState;
  namespaces: NamespacesState;
  publicKeys: PublicKeysState;
  sessions: SessionsState;
  snackbar: SnackbarState;
  spinner: SpinnerState;
  stats: StatsState;
  announcement: AnnouncementState;
  instance: InstanceState;
}

// eslint-disable-next-line symbol-description
export const key: InjectionKey<Store<State>> = Symbol();

export const store = createStore<State>({
  modules: {
    users,
    auth,
    devices,
    firewallRules,
    layout,
    license,
    namespaces,
    publicKeys,
    sessions,
    snackbar,
    spinner,
    stats,
    announcement,
    instance,
  },
});

export function useStore(): Store<State> {
  return vuexUseStore(key);
}
