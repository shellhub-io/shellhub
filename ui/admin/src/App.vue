<template>
  <v-app>
    <component :is="layout" :data-test="`${layout}-component`" />
  </v-app>
</template>

<script lang="ts">
import { defineComponent, onMounted, computed } from "vue";
import { useRouter } from "vue-router";
import SimpleLayout from "./layouts/SimpleLayout.vue";
import AppLayout from "./layouts/AppLayout.vue";
import { useStore } from "./store";
import { INotificationsError } from "./interfaces/INotifications";

export default defineComponent({
  name: "App",
  components: {
    appLayout: AppLayout,
    simpleLayout: SimpleLayout,
  },
  setup() {
    const store = useStore();
    const router = useRouter();

    const layout = computed(() => store.getters["layout/getLayout"]);

    const isLoggedIn = computed(() => store.getters["auth/isLoggedIn"]);

    const currentRoute = computed(() => router.currentRoute.value.path);

    onMounted(async () => {
      if (!isLoggedIn.value) {
        try {
          await store.dispatch("auth/logout");
          store.dispatch("layout/setLayout", "simpleLayout");
          router.push("/login");
        } catch {
          store.dispatch("snackbar/showSnackbarErrorAction", INotificationsError.namespaceLoad);
        }
      }

      if (isLoggedIn.value && currentRoute.value !== "/login") {
        const license = await store.dispatch("license/get");
        if (!license || license.expired) {
          store.dispatch("snackbar/showSnackbarErrorAction", INotificationsError.license);
          store.dispatch("layout/setLayout", "appLayout");
          router.push("/license");
        }
        store.dispatch("layout/setLayout", "appLayout");
      }
    });

    return {
      layout,
    };
  },
});
</script>
