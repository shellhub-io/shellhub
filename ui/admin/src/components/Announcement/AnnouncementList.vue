<template>
  <DataTable
    :headers="headers"
    :items="announcements"
    :itemsPerPage="itemsPerPage"
    :nextPage="next"
    :previousPage="prev"
    :loading="loading"
    :totalCount="numberAnnouncements"
    :page="page"
    :actualPage="page"
    @changeItemsPerPage="changeItemsPerPage"
    @clickNextPage="next"
    @clickPreviousPage="prev"
    data-test="announcement-list"
  >
    <template v-slot:rows>
      <tr v-for="(announcement, index) in announcements" :key="index">
        <td class="text-left" data-test="announcement-uuid">
          <v-chip>
            {{ announcement.uuid }}
          </v-chip>
        </td>
        <td class="text-left" data-test="announcement-title">
          {{ announcement.title }}
        </td>
        <td class="text-left">
          {{ formatDate(announcement.date) }}
        </td>
        <td class="text-left" data-test="announcement-actions">
          <v-tooltip bottom anchor="bottom">
            <template v-slot:activator="{ props }">
              <v-icon
                tag="a"
                dark
                v-bind="props"
                @click="redirectToAnnouncement(announcement)"
                @keyup.enter="redirectToAnnouncement(announcement)"
                tabindex="0"
              >mdi-information
              </v-icon>
            </template>
            <span>Info</span>
          </v-tooltip>

          <AnnouncementEdit :announcement="announcement" @update="refreshAnnouncements" />

          <AnnouncementDelete :uuid="announcement.uuid" @update="refreshAnnouncements" />
        </td>
      </tr>
    </template>
  </DataTable>
</template>

<script lang="ts">
import { computed, defineComponent, onMounted, ref, watch } from "vue";
import { useRouter } from "vue-router";
import moment from "moment";
import { useStore } from "../../store";
import { INotificationsError } from "../../interfaces/INotifications";
import DataTable from "../DataTable.vue";
import displayOnlyTenCharacters from "../../hooks/string";
import AnnouncementDelete from "./AnnouncementDelete.vue";
import AnnouncementEdit from "./AnnouncementEdit.vue";
import { IAnnouncements } from "../../interfaces/IAnnouncements";

export default defineComponent({
  name: "AnnouncementList",
  setup() {
    const store = useStore();
    const router = useRouter();
    const itemsPerPage = ref(10);
    const loading = ref(false);
    const page = ref(1);

    const getAnnouncements = async (
      perPagaeValue: number,
      pageValue: number,
    ) => {
      try {
        loading.value = true;
        const hasAnnouncements = await store.dispatch(
          "announcement/getAnnouncements",
          {
            perPage: perPagaeValue,
            page: pageValue,
          },
        );
        if (!hasAnnouncements && page.value > 1) {
          page.value--;
        }
        loading.value = false;
      } catch (error) {
        store.dispatch(
          "snackbar/showSnackbarErrorAction",
          INotificationsError.announcementList,
        );
      }
    };
    onMounted(async () => {
      try {
        loading.value = true;
        getAnnouncements(itemsPerPage.value, page.value);
      } catch (error) {
        store.dispatch(
          "snackbar/showSnackbarErrorAction",
          INotificationsError.announcementList,
        );
      } finally {
        loading.value = false;
      }
    });
    const next = async () => {
      await getAnnouncements(itemsPerPage.value, ++page.value);
    };
    const prev = async () => {
      try {
        if (page.value > 1) await getAnnouncements(itemsPerPage.value, --page.value);
      } catch (error) {
        store.dispatch(
          "snackbar/showSnackbarErrorAction",
          INotificationsError.announcementList,
        );
      }
    };
    const changeItemsPerPage = async (newItemsPerPage: number) => {
      itemsPerPage.value = newItemsPerPage;
    };
    watch(itemsPerPage, async () => {
      await getAnnouncements(itemsPerPage.value, page.value);
    });
    const announcements = computed(
      () => store.getters["announcement/announcements"],
    );
    const numberAnnouncements = computed(
      () => store.getters["announcement/numberAnnouncements"],
    );

    const refreshAnnouncements = async () => {
      await getAnnouncements(itemsPerPage.value, page.value);
    };

    const formatDate = (date: string) => moment(date).format("LL");

    const redirectToAnnouncement = (announcement: IAnnouncements) => {
      router.push({
        name: "announcementDetails",
        params: { uuid: announcement.uuid },
      });
    };

    return {
      headers: [
        {
          text: "Id",
          value: "uuid",
          align: "left",
        },
        {
          text: "Title",
          value: "title",
          align: "left",
        },
        {
          text: "Date",
          value: "date",
          align: "left",
        },
        {
          text: "Actions",
          value: "actions",
          align: "left",
        },
      ],
      announcements,
      numberAnnouncements,
      loading,
      itemsPerPage,
      page,
      next,
      prev,
      changeItemsPerPage,
      displayOnlyTenCharacters,
      formatDate,
      refreshAnnouncements,
      redirectToAnnouncement,
    };
  },
  components: { DataTable, AnnouncementDelete, AnnouncementEdit },
});
</script>
