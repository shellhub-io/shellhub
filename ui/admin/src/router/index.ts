import { createRouter, createWebHistory } from "vue-router";
import Login from "@admin/views/Login.vue";
import Dashboard from "@admin/views/Dashboard.vue";
import Users from "@admin/views/Users.vue";
import SettingsLicense from "@admin/components/Settings/SettingsLicense.vue";
import SettingsAuthentication from "@admin/components/Settings/SettingsAuthentication.vue";
import Namespaces from "@admin/views/Namespaces.vue";
import Settings from "@admin/views/Settings.vue";

import { store } from "../store";

const routes = [
  {
    path: "/login",
    name: "login",
    component: Login,
  },
  {
    path: "/",
    name: "dashboard",
    component: Dashboard,
  },
  {
    path: "/users",
    name: "users",
    component: Users,
  },
  {
    path: "/user/:id",
    name: "userDetails",
    component: () => import("@admin/views/UserDetails.vue"),
  },
  {
    path: "/devices",
    name: "devices",
    component: () => import("@admin/views/Device.vue"),
    redirect: {
      name: "listDevices",
    },
    children: [
      {
        path: "",
        name: "listDevices",
        component: () => import("@admin/views/Device.vue"),
      },
    ],
  },
  {
    path: "/device/:id",
    name: "deviceDetails",
    component: () => import("@admin/views/DeviceDetails.vue"),
  },
  {
    path: "/sessions",
    name: "sessions",
    component: () => import("@admin/views/Sessions.vue"),
  },
  {
    path: "/session/:id",
    name: "sessionDetails",
    component: () => import("@admin/views/SessionDetails.vue"),
  },
  {
    path: "/settings",
    name: "Settings",
    component: Settings,
    redirect: { name: "SettingProfile" },
    children: [
      {
        path: "authentication",
        name: "SettingAuthentication",
        component: SettingsAuthentication,
      },
      {
        path: "license",
        name: "SettingLicense",
        component: SettingsLicense,
      },
    ],
  },
  {
    path: "/firewall-rules",
    name: "firewall-rules",
    component: () => import("@admin/views/FirewallRules.vue"),
  },
  {
    path: "/firewall-rules/:id",
    name: "firewallRulesDetails",
    component: () => import("@admin/views/FirewallRulesDetails.vue"),
  },
  {
    path: "/namespaces",
    name: "namespaces",
    component: Namespaces,
  },
  {
    path: "/namespace/:id",
    name: "namespaceDetails",
    component: () => import("@admin/views/NamespaceDetails.vue"),
  },
  {
    path: "/announcements",
    name: "announcements",
    component: () => import("@admin/views/Announcements.vue"),
  },
  {
    path: "/announcement/:uuid",
    name: "announcementDetails",
    component: () => import("@admin/views/AnnouncementDetails.vue"),
  },
  {
    path: "/new-announcement",
    name: "new-announcement",
    component: () => import("@admin/views/NewAnnouncement.vue"),
  },
];

const router = createRouter({
  history: createWebHistory("/admin/"),
  routes,
});

router.beforeEach((to, from, next) => {
  if (to.path !== "/login") {
    if (store.getters["auth/isLoggedIn"]) {
      return next();
    }
    return next(`/login?redirect=${to.path}`);
  }
  if (store.getters["auth/isLoggedIn"]) {
    if (to.path === "/login" && to.query.token) {
      return next();
    }
    return next("/");
  }

  return next();
});

export default router;
