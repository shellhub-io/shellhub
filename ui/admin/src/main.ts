import { createApp } from "vue";
import vuetify from "@ui/plugins/vuetify";
import App from "./App.vue";
import { loadFonts } from "./plugins/webfontloader";
import { key, store } from "./store";
import router from "./router";
import SnackbarComponent from "./components/Snackbar/Snackbar.vue";

const app = createApp(App);

loadFonts();

app.use(vuetify);
app.use(router);
app.use(store, key);
app.component("SnackbarComponent", SnackbarComponent);
app.mount("#app");
