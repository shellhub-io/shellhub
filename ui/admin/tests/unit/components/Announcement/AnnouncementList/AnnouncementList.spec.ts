import { createStore } from "vuex";
import { createVuetify } from "vuetify";
import { mount, VueWrapper } from "@vue/test-utils";
import { beforeEach, describe, expect, it, vi } from "vitest";
import AnnouncementList from "../../../../../src/components/Announcement/AnnouncementList.vue";
import { key } from "../../../../../src/store";
import routes from "../../../../../src/router";

type AnnouncementListWrapper = VueWrapper<InstanceType<typeof AnnouncementList>>;

const announcements = [
  {
    uuid: "eac7e18d-7127-41ca-b68b-8242dfdbaf4c",
    title: "Announcement 1",
    content: "## ShellHub new features \n - New feature 1 \n - New feature 2 \n - New feature 3",
    date: "2022-12-15T19:45:45.618Z",
  },
  {
    uuid: "eac7e18d-7127-41ca-b68b-8242dfdbaf5b",
    title: "Announcement 2",
    content: "## ShellHub new features \n - New feature 1 \n - New feature 2 \n - New feature 3",
    date: "2022-12-15T19:45:45.618Z",
  },
];

const numberAnnouncements = 2;

describe("Announcement List", () => {
  const store = createStore({
    state: {
      announcements,
    },
    getters: {
      "announcement/announcements": () => announcements,
      "announcement/numberAnnouncements": () => numberAnnouncements,
    },
    actions: {
      "announcement/getAnnouncements": vi.fn(),
      "announcement/announcements": vi.fn(),
      "snackbar/showSnackbarErrorAction": vi.fn(),
    },
  });

  const vuetify = createVuetify();

  let wrapper: AnnouncementListWrapper;

  beforeEach(() => {
    wrapper = mount(AnnouncementList, {
      global: {
        plugins: [[store, key], vuetify, routes],
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper).toBeTruthy();
  });

  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Renders the correct data", () => {
    expect(wrapper.vm.itemsPerPage).toBe(10);
    expect(wrapper.vm.page).toBe(1);
    expect(wrapper.vm.loading).toBe(false);
  });

  it("Renders the correct computed", () => {
    expect(wrapper.vm.numberAnnouncements).toBe(numberAnnouncements);
    expect(wrapper.vm.announcements).toBe(announcements);
  });

  it("Renders the correct HTML", () => {
    expect(wrapper.find("[data-test='announcement-list']").exists()).toBeTruthy();
    expect(wrapper.find("[data-test='announcement-uuid']").exists()).toBeTruthy();
    expect(wrapper.find("[data-test='announcement-title']").exists()).toBeTruthy();
    expect(wrapper.find("[data-test='announcement-actions']").exists()).toBeTruthy();
  });
});
