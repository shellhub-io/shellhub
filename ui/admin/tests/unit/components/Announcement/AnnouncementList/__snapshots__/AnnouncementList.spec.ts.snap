// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Announcement List > Renders the component 1`] = `
"<div data-v-90b5158f="" page="1" data-test="announcement-list">
  <div data-v-90b5158f="" class="v-table v-theme--light v-table--density-default">
    <!---->
    <div class="v-table__wrapper">
      <table>
        <thead data-v-90b5158f="">
          <tr data-v-90b5158f="">
            <th data-v-90b5158f="" class="text-left"><span data-v-90b5158f="">Id</span></th>
            <th data-v-90b5158f="" class="text-left"><span data-v-90b5158f="">Title</span></th>
            <th data-v-90b5158f="" class="text-left"><span data-v-90b5158f="">Date</span></th>
            <th data-v-90b5158f="" class="text-left"><span data-v-90b5158f="">Actions</span></th>
          </tr>
        </thead>
        <tbody data-v-90b5158f="">
          <tr>
            <td class="text-left" data-test="announcement-uuid"><span class="v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal" draggable="false"><!----><span class="v-chip__underlay"></span>
              <!---->
              <!---->
              <div class="v-chip__content" data-no-activator="">eac7e18d-7127-41ca-b68b-8242dfdbaf4c</div>
              <!---->
              <!----></span>
            </td>
            <td class="text-left" data-test="announcement-title">Announcement 1</td>
            <td class="text-left">December 15, 2022</td>
            <td class="text-left" data-test="announcement-actions"><a class="mdi-information mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" dark="" aria-describedby="v-tooltip-1"></a>
              <!--teleport start-->
              <!--teleport end--><a class="mdi-pencil mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" dark="" aria-describedby="v-tooltip-2" aria-label="Dialog edit announcement"></a>
              <!--teleport start-->
              <!--teleport end-->
              <!---->
              <!--teleport start-->
              <!--teleport end--><a class="mdi-delete mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" dark="" aria-describedby="v-tooltip-7"></a>
              <!--teleport start-->
              <!--teleport end-->
              <!---->
              <!---->
            </td>
          </tr>
          <tr>
            <td class="text-left" data-test="announcement-uuid"><span class="v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal" draggable="false"><!----><span class="v-chip__underlay"></span>
              <!---->
              <!---->
              <div class="v-chip__content" data-no-activator="">eac7e18d-7127-41ca-b68b-8242dfdbaf5b</div>
              <!---->
              <!----></span>
            </td>
            <td class="text-left" data-test="announcement-title">Announcement 2</td>
            <td class="text-left">December 15, 2022</td>
            <td class="text-left" data-test="announcement-actions"><a class="mdi-information mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" dark="" aria-describedby="v-tooltip-9"></a>
              <!--teleport start-->
              <!--teleport end--><a class="mdi-pencil mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" dark="" aria-describedby="v-tooltip-10" aria-label="Dialog edit announcement"></a>
              <!--teleport start-->
              <!--teleport end-->
              <!---->
              <!--teleport start-->
              <!--teleport end--><a class="mdi-delete mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" dark="" aria-describedby="v-tooltip-15"></a>
              <!--teleport start-->
              <!--teleport end-->
              <!---->
              <!---->
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <!---->
  </div>
  <hr data-v-90b5158f="" class="v-divider v-theme--light" aria-orientation="horizontal" role="separator">
  <!--v-if-->
  <div data-v-90b5158f="" class="d-flex w-100 justify-end align-center"><span data-v-90b5158f="" class="text-subtitle-2 mr-4">Items per page:</span>
    <div data-v-90b5158f="">
      <div data-v-90b5158f="" class="v-input v-input--horizontal v-input--density-default v-theme--light v-locale--is-ltr v-input--dirty v-text-field v-input--plain-underlined v-combobox v-combobox--single mb-4">
        <!---->
        <div class="v-input__control">
          <div class="v-field v-field--active v-field--appended v-field--dirty v-field--no-label v-field--variant-underlined v-theme--light v-locale--is-ltr" role="combobox" aria-haspopup="menu" aria-expanded="false" aria-controls="v-menu-18">
            <div class="v-field__overlay"></div>
            <div class="v-field__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                <!---->
                <div class="v-progress-linear__background"></div>
                <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__indeterminate">
                    <div class="v-progress-linear__indeterminate long"></div>
                    <div class="v-progress-linear__indeterminate short"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <div class="v-field__field" data-no-activator="">
              <!---->
              <!---->
              <!---->
              <div class="v-field__input" data-no-activator="">
                <!---->
                <!---->
                <div class="v-combobox__selection"><span class="v-combobox__selection-text">10<!----></span></div><input size="1" type="text" id="input-16" aria-describedby="input-16-messages" outlined="" value="10">
              </div>
              <!---->
            </div>
            <!---->
            <div class="v-field__append-inner">
              <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable v-combobox__menu-icon" role="button" aria-hidden="false" tabindex="-1" aria-label="Open" title="Open"></i>
              <!---->
            </div>
            <div class="v-field__outline">
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <!---->
        <!---->
      </div>
    </div>
    <div data-v-90b5158f="" class="d-flex align-center"><button data-v-90b5158f="" type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain" disabled=""><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-left mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
        <!---->
        <!---->
      </button><span data-v-90b5158f="" class="text-subtitle-2">1 of 1</span><button data-v-90b5158f="" type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain" disabled=""><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-right mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
        <!---->
        <!---->
      </button></div>
  </div>
</div>"
`;
